### ISSUE TITLE
Add Aqara Acpartner v3 (Advanced version) support

### ISSUE THREAD

**Issue by cxlwill** [Danielhiversen/PyXiaomiGateway#65]
Related to [this issue](https://github.com/syssi/xiaomi_airconditioningcompanion/issues/6) .
Accoring to [Aqara's dev docs](http://docs.opencloud.aqara.cn/development/en-gateway-LAN-communication/#encryption-mechanism), `acpartner.v3` officially support Gateway function. Here's debug log at HA when I try to add it:
```
2018-03-25 20:01:06 DEBUG (Thread-23) [xiaomi_gateway] MCAST (heartbeat) << {'model': 'acpartner.v3', 'sid': '7811dcb2a7d9', 'token': 'nZMYm8LUp0SwNAcc', 'params': [{'ip': '192.168.50.99'}], 'cmd': 'heartbeat'}
2018-03-25 20:01:06 ERROR (Thread-16) [xiaomi_gateway] No data in response from hub {'model': 'acpartner.v3', 'sid': '7811dcb2a7d9', 'token': 'nZMYm8LUp0SwNAcc', 'params': [{'ip': '192.168.50.99'}], 'cmd': 'heartbeat'}
```
`namp` test:
```
sudo nmap -sU -p 9898 192.168.50.99
Starting Nmap 7.40 ( https://nmap.org ) at 2018-03-25 16:53 CST
Nmap scan report for lumi-acpartner-v3_miio78478387 (192.168.50.99)
Host is up (0.020s latency).
PORT     STATE SERVICE
9898/udp open  monkeycom
MAC Address: 78:11:DC:B2:A7:D9 (Unknown)
```
`nc` test:
```
nc -v -uz 192.168.50.99 9898
found 0 associations
found 1 connections:
     1:	flags=82<CONNECTED,PREFERRED>
	outif (null)
	src 192.168.50.143 port 57240
	dst 192.168.50.99 port 9898
	rank info not available

Connection to 192.168.50.99 port 9898 [udp/monkeycom] succeeded!
```
Which means that Port 9898 is open at this device.

I use UDP/TCP tool to test JSON multicast and unicast like `{"cmd":"discovery"}` and `{"cmd":"whois"}` both got positive result. It reported itself and sub-devices like normal `gateway`.

Except illumination sensor and light(No hardware), it works like other `gateway` including ringtone. 

I think it's time finally to expand this library to cover this device. According to docs, right now only v3 support this function. So previous v2 and v1 have to wait for Aqara to update firmware.

**Comment by syssi** [comment]
I will care about the Integration / updating the docs.

**Comment by syssi** [comment]
Fixed by #66.
